packages/kbn-lens-embeddable-utils/config_builder/types.ts

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { FormulaPublicApi, TypedLensByValueInput } from '@kbn/lens-plugin/public';
+import { Filter, Query } from '@kbn/es-query';
+import { DataViewsPublicPluginStart } from '@kbn/data-views-plugin/public';
+import { Datatable } from '@kbn/expressions-plugin/common';
+
+export type LensAttributes = TypedLensByValueInput['attributes'];
+export const DEFAULT_LAYER_ID = 'layer_0';
+
+type Identity<T> = T extends object
+  ? {
+      [P in keyof T]: T[P];
+    }
+  : T;
+
+export type ChartType =
+  | 'xy'
+  | 'pie'
+  | 'heatmap'
+  | 'metric'
+  | 'gauge'
+  | 'donut'
+  | 'mosaic'
+  | 'regionmap'
+  | 'table'
+  | 'tagcloud'
+  | 'treemap';
+
+export interface TimeRange {
+  from: string;
+  to: string;
+  type: 'relative' | 'absolute';
+}
+
+export type LensLayerQuery = string;
+export interface LensDataviewDataset {
+  index: string;
+  timeFieldName?: string;
+}
+
+export type LensDatatableDataset = Datatable;
+
+export interface LensESQLDataset {
+  esql: string;
+}
+
+export type LensDataset = LensDataviewDataset | LensDatatableDataset | LensESQLDataset;
+
+export interface LensBaseConfig {
+  title: string;
+  /** default data view id or index pattern to use, it can be overriden on each query */
+  dataset?: LensDataset;
+}
+
+export interface LensBaseLayer {
+  label: string;
```

Comment:
If I am not mistaken the label in the layer is used only for saved annotations that we don't support. So I think this label should be optional param?